src/NuGet.Core/NuGet.LibraryModel/LibraryDependency.cs

```diff
@@ -50,8 +51,16 @@ public class LibraryDependency : IEquatable<LibraryDependency>
         /// <summary>Initializes a new instance of the LibraryDependency class.</summary>
         /// <remarks>Required properties must be set when using this constructor.</remarks>
         public LibraryDependency()
+            : this(new List<NuGetLogCode>())
```

Comment:
Can this simply be `Array.Empty<NuGetLogCode>()` so that we can avoid invoking the new constructor by passing an empty collection in so many places?

Reply:
Unfortunately no, if you have an existing code path that does this:

```c#
var libraryDependency = new LibraryDependency();

libraryDependency.NoWarn.Add(NuGetLogCode.NU1000);
```

it will break because the list is empty and when you call `Add()` you get an exception about no capacity.  So I left this constructor in place but all of NuGet's code uses the other constructor to signal if the NoWarn codes should be an exact list it specifies or an empy list that it won't be adding to.

Reply:
We could potentially address the above limitation by making `NoWarn` a read-only collection. Is it because of public API requirements that we do not want to change the types?

Reply:
Yes, I was trying to avoid making a breaking change.  We could make a breaking change later, for this change I'm just trying to address unnecessary memory allocations.